### ISSUE TITLE
Add tutorial levels back

### ISSUE THREAD

**Issue by Wuzzy2** [SimonN/LixD#336]
Would it be possible to add the old tutorial levels back into the game?

I have also contributed a few tutorial levels in the old forums (asdfasdf), do you remember (user name “Wuzzy”)?

**Comment by stavrossk** [comment]
Yes, I would love to see some tutorial levels, especially to introduce advanced features and skills.

**Comment by SimonN** [comment]
This is tricky because it touches the game's overall design.

Yes, I remember the extra tutorials from asdfasdf. In 2015-2017, I decided to remove all tutorials in favor of easier maps in lemforum/Lovely and lemforum/Quirky. Do these easy maps teach enough?

Three possibilities:
* No tutorials
* Textless tutorials
* Textful tutorials

Ideally, the game teaches everything without words. Can we teach advanced tricks without words, by designing very small maps that require only one single trick? Or would advanced tutorials spoil tricks in harder maps?

Can we get by without printing level-specific text during play? Sadly, the physics are complicated and inconsistent, and need either explanations or many showcases: Walkers either cancel or turn. You can assign walker to blocker and imploder to blocker, but only imploder to climber (not walker to climber). Nothing here is discoverable, player is supposed to try everything.

I'd really like to avoid text, and teach everything with showcase levels. If we avoid text, we need many textless tutorial levels. That would be okay with me.

Clam/Outtakes has tutorial-ish levels, such as the Turnaround Compendium. Rubix/Pleasant has nice easy levels, such as A Soft Landing (cube under the deadly fall) and Alternative Cancellation (many ways to cancel diggers).

What should a new player play first? Long playground maps with many skills (several people like that, I don't), or concise single-trick levels (I like that)?

**Comment by Wuzzy2** [comment]
Well, there are various gotchas in the game:

* How each of the skills works
* **Exact** (!) purpose of walker (you already mentioned)
* How to force left/right
* Miner/basher turns around when hitting steel
* Builder turns around when colliding, but not platformer
* Difference between imploder and exploder
* Some traps are not active all the time
* You sometimes must nuke your lixes to complete a level (to get rid of blockers)

In some levels, you *must* know these minor details, otherwise they are unsolvable. These are also not part of puzzle solving, but just the game rules.

Some of there, like forcing left/right, are impossible to teach without words.
The walker is also complex enough to probably require words (sorry). This is because walker and left/right is about the game controls. It's unfair to expect from the player to figure out the correct controls by trial and error.

The purpose of tutorials is to teach the basic *rules of the game*, but not more. I.e. no tricks which the player could logically derive from the rules. For example, a basic rule is “a lix falls down when it loses ground below its feet”. Another one is “a blocker stays still”. A trick derived from the rules would be “to make a blocker walk again, remove ground below the blocker”.

I think it is generally better to separate the tutorial levels from regular levels. It's not good when crucial rules of the games are tought in levels scattered randomly around the game. Also, those levels rarely introduce the new concepts in isolation, so they are not really a replacement.

Basically, the old tutorial already offered all of that. Note that some of the old tutorials are quite fun, especially the exploders and the batters. :)

Text-less tutorials are do-able, but only to a limited extent. Maybe the old tutorial levels could just be turned into a level set with a difficulty lower than Lovely. I like the idea of exposing each skill separately, before throwing the player into “real” levels.

FYI: These are the old tutorials (not all skills are covered):
https://github.com/SimonN/Lix/tree/master/levels/tutorial/skills
I can send you the levels for the missing skills, if you want.

> What should a new player play first? Long playground maps with many skills (several people like that, I don't), or concise single-trick levels (I like that)?

Definitely concise levels, one skill at a time. Any Way You Want (first level of lemforum/Lovely) throws the player into the game which a whopping 14 skills, and a fall hazard. This might be too overwhelming for complete newbies, it's better to start slow.

Even if you are completely opposed to a tutorial, we could add a level set with a difficulty lower than Lovely. It disguises itself as a normal level set but is so easy and introduces new concepts so slowly that it's basically a tutorial in disguise. :)

**Comment by Wuzzy2** [comment]
Well in C++ Lix the tutorial levels already worked that way. I believe many tutorial levels can be reused almost verbatim.

**Comment by SimonN** [comment]
Since Lix 0.9.25 from 2 months ago, Lix has tutorials for the basic skills.

We should still add more advanced tutorials, therefore I'll leave this issue open.

I should still implement tutorial messages. They should be displayed throughout the tutorial. They should be translatable that are displayed throughout  and can be translated.
